great_expectations/rule_based_profiler/config/base.py

```diff
@@ -427,39 +427,44 @@ def resolve_config_using_acceptable_arguments(
         Returns:
             An instance of RuleBasedProfilerConfig that represents the reconciled profiler.
         """
-        runtime_config: RuleBasedProfilerConfig = profiler.config
-
-        runtime_variables: dict = profiler.reconcile_profiler_variables_as_dict(
-            variables=variables
+        effective_variables: Optional[
+            ParameterContainer
+        ] = profiler.reconcile_profiler_variables(
+            variables=variables,
         )
-        variables_container: ParameterContainer = (
-            build_parameter_container_for_variables(variables_configs=runtime_variables)
+        runtime_variables: Optional[Dict[str, Any]] = convert_variables_to_dict(
+            variables=effective_variables
+        )
+
+        effective_rules: List["Rule"] = profiler.reconcile_profiler_rules(  # noqa: F821
+            rules=rules,
         )
 
-        effective_rules: Dict[
-            str, "Rule"  # noqa: F821
-        ] = profiler.reconcile_profiler_rules_as_dict(rules=rules)
-        runtime_rules: Dict[str, dict] = {}
-        for name, rule in effective_rules.items():
-            rule_with_substituted_vars: dict = (
-                RuleBasedProfilerConfig._substitute_variables_in_config(
-                    rule=rule, variables_container=variables_container
-                )
+        rule: "Rule"  # noqa: F821
+        effective_rules_dict: Dict[str, "Rule"] = {  # noqa: F821
+            rule.name: rule for rule in effective_rules
+        }
+        runtime_rules: Dict[str, dict] = {
+            name: RuleBasedProfilerConfig._substitute_variables_in_config(
+                rule=rule,
+                variables_container=effective_variables,
             )
-            runtime_rules[name] = rule_with_substituted_vars
+            for name, rule in effective_rules_dict.items()
+        }
```

Comment:
I'd request we keep this a for loop and remove the comprehension. It greatly helps with debugging and is easier to read.

Reply:
This is an acceptable usage of a comprehension but I'd request that we use only use it in cases that it makes sense.

There are other instances in our codebase that are wildly complicated and unwieldy. It may be elegant but readability should be a higher priority in my opinion.

Reply:
@cdkini I agree that we should think twice before going ahead with using complex (not simple) double-comprehensions and even single ones.  The question becomes what is complex.  To me the barrier becomes when the right hand side is some complicated method with many arguments, but I admit it is a bit of a fuzzy boundary.  I know that a triple-comprehension is too much for me, but most double ones can be rationalized as being on the elegant side.  Not sure.  I do find, however, some single comprehensions, if "institutionalized" in a given module, become idiomatic, which I then want to copy for consistency purposes.  Happy to discuss.  Thanks.

Reply:
@alexsherstinsky I do see your point, but also find myself doing a bit of mental gymnastics here. If we are going to keep the code, could we add an inline comment?

Reply:
@Shinnnyshinshin Sorry, just saw this -- for the benefit of the release, let us add the comment later.